app/common/data/interfaces/collections.py

```diff
@@ -61,6 +62,16 @@ def update_collection_schema(schema: CollectionSchema, *, name: str) -> Collecti
     return schema
 
 
+def update_collection_data(
+    collection: Collection, question: Question, data: str | int | float | bool | None
+) -> Collection:
+    collection.data = copy.deepcopy(collection.data)
```

Comment:
In my WIP branch on this, I did just pull in `sqlalchemy-json`. I think let's just do that now to avoid needing to do this.